app/mailers/registration_mailer.rb

```diff
@@ -10,9 +10,9 @@ def welcome_email(user, password=nil)
 
   def duplicate_email_detection(user, duplicate)
     @user = user
-    @duplicate_provider = duplicate.provider? ? duplicate.provider.camelcase : I18n.t(:mail_duplicate_email_exists_email)
-    @user_provider = user.provider? ? user.provider.camelcase : I18n.t(:mail_duplicate_email_exists_email)
-    @mail_body = (duplicate.provider? || user.provider?) ? :mail_duplicate_email_exists_body : :mail_duplicate_email_exists_body_both_emails
+    @duplicate_provider = duplicate.encrypted_password? ? I18n.t(:mail_duplicate_email_exists_email) : duplicate.get_social_accounts_for_login.first.provider
```

Comment:
Does this logic allow a duplicate email for the SAME user?

Reply:
No just need to catch a provider for duplicate email in case user tried to register with existing email to use it on email body